### PR TITLE
fix(helper): backslash escaping (\s\S) in template strings are handled incorrectly

### PR DESCRIPTION
**Pull request by dbabaioff**
This fixes a issue with the selector with attrs RegExp and template string (https://github.com/angular/universal/commit/787d98c90dfdb713c1e80a97764c51424ade967b)

Before:
```js
var regExpSelect = "<" + escapeRegExp(selector) + "[^>]*>([sS]*?)</" + escapeRegExp(selector) + ">";
```

After:
```js
var regExpSelect = "<" + escapeRegExp(selector) + "[^>]*>([\\s\\S]*?)</" + escapeRegExp(selector) + ">";
```

See: https://github.com/Microsoft/TypeScript/issues/2803

### CODE REVIEW

**Comment by dbabaioff** [comment]
(Committed a fix for: https://github.com/angular/universal/issues/142 on the same PR - wanted to open a separate PR)
Closes: https://github.com/angular/universal/issues/142

File: modules/universal/server/src/platform/node.ts [hunk 1]
```diff
@@ -55,6 +58,41 @@ export function initNodeAdapter() {
...
 
+export class NodeXHRImpl extends XHR {
+  get(templateUrl: string): Promise<string> {
+    let completer: PromiseCompleter<string> = PromiseWrapper.completer(),
+      parsedUrl = url.parse(templateUrl);
+
+    http.get(templateUrl, (res) => {
+      res.setEncoding('utf8');
+
+      // normalize IE9 bug (http://bugs.jquery.com/ticket/1450)
```
Comment (gdi2290) [inline modules/universal/server/src/platform/node.ts]: lol we don't need to do this since we're in node.js environment but don't worry I'll merge it in

**Comment by gdi2290** [comment]
LGTM, merging

File: modules/universal/server/src/platform/node.ts (hunk 1 above)
Comment (dbabaioff) [inline modules/universal/server/src/platform/node.ts]: We still need to create a class for xhr (node.js), no?

Comment (gdi2290) [inline modules/universal/server/src/platform/node.ts]: I was talking about the IE9 comment

**Pull request merged**